sources/engine/Stride.Graphics/Semantic/ISemantic.cs

```diff
@@ -0,0 +1,176 @@
+// Copyright (c) .NET Foundation and Contributors (https://dotnetfoundation.org/ & https://stride3d.net)
+// Distributed under the MIT license. See the LICENSE.md file in the project root for more information.
+
+#nullable enable
+namespace Stride.Graphics.Semantic;
```

Comment:
What about naming the namespace in plural (`Stride.Graphics.Semantics`)?
That way it conveys at a glance that it contains semantic definitions, and it won't be confused with a type.